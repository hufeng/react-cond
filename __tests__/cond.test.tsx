import React from 'react';
import renderer from 'react-test-renderer';
import Cond from '../src/cond/';
const { Case, Default } = Cond;

describe('cond test suite', () => {
  it('cond no children', () => {
    const CondEmpty = () => <Cond />;

    const component = renderer.create(<CondEmpty />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('cond one children test is true', () => {
    const CondOne = () =>
      <Cond>
        <Case test={1 == 1} component={<div>I am match</div>} />
      </Cond>;

    const component = renderer.create(<CondOne />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('cond one chidld test is false', () => {
    const CondOne = () =>
      <Cond>
        <Case test={1 != 1} component={<div>I am match</div>} />
      </Cond>;

    const component = renderer.create(<CondOne />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('cond multiple test ', () => {
    const MulCond = () =>
      <Cond>
        <Case test={1 != 1} component={<div>I am match1</div>} />
        <Case test={1 != 1} component={<div>I am match2</div>} />
        <Case test={1 != 1} component={<div>I am match3</div>} />
        <Case test={1 != 1} component={<div>I am match4</div>} />
        <Case test={1 == 1} component={<div>I am match5</div>} />
      </Cond>;

    const component = renderer.create(<MulCond />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('cond one default', () => {
    const CondDefault = () =>
      <Cond>
        <Default component={<div>I Am Default</div>} />
      </Cond>;

    const component = renderer.create(<CondDefault />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('cond case and default', () => {
    const MulCond1 = () =>
      <Cond>
        <Case test={1 != 1} component={<div>I am match1</div>} />
        <Case test={1 != 1} component={<div>I am match2</div>} />
        <Case test={1 != 1} component={<div>I am match3</div>} />
        <Case test={1 != 1} component={<div>I am match4</div>} />
        <Case test={1 == 1} component={<div>I am match5</div>} />
        <Default component={<div>I am default</div>} />
      </Cond>;

    const MulCond2 = () =>
      <Cond>
        <Case test={1 != 1} component={<div>I am match1</div>} />
        <Case test={1 != 1} component={<div>I am match2</div>} />
        <Case test={1 != 1} component={<div>I am match3</div>} />
        <Case test={1 != 1} component={<div>I am match4</div>} />
        <Case test={1 != 1} component={<div>I am match5</div>} />
        <Default component={<div>I am default</div>} />
      </Cond>;

    const component = renderer.create(<MulCond1 />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const component2 = renderer.create(<MulCond2 />);
    const tree2 = component2.toJSON();
    expect(tree2).toMatchSnapshot();
  });

  it('modify default props is error', () => {
    // const MulCond = () =>
    //   <Cond>
    //     <Case test={1 != 1} component={<div>I am match1</div>} />
    //     <Default test={1 != 1} component={<div>I am match3</div>} />
    //   </Cond>;
    // try {
    //   const component = renderer.create(<MulCond />);
    // } catch (err) {
    //   expect(err).toBe(
    //     new Error('Could not modify default test props is false')
    //   );
    // }
  });
});
