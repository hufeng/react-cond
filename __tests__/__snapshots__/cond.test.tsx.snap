// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cond test suite cond case and default 1`] = `
<div>
  I am match5
</div>
`;

exports[`cond test suite cond case and default 2`] = `
<div>
  I am default
</div>
`;

exports[`cond test suite cond multiple test  1`] = `
<div>
  I am match5
</div>
`;

exports[`cond test suite cond no children 1`] = `null`;

exports[`cond test suite cond one chidld test is false 1`] = `null`;

exports[`cond test suite cond one children test is true 1`] = `
<div>
  I am match
</div>
`;

exports[`cond test suite cond one default 1`] = `
<div>
  I Am Default
</div>
`;
